VSharp.SILI/DecompilerServices.fs

```diff
@@ -160,8 +190,10 @@ module internal DecompilerServices =
         let regularFields = decompiledClass.Fields |> Seq.filter (isDecompiledFieldStatic isStatic) |> Seq.map extractDecompiledFieldInfo |> List.ofSeq in
         let backingFields = decompiledClass.Properties |> Seq.filter (isStaticBackingField isStatic) |> Seq.map extractBackingFieldInfo |> List.ofSeq in
         let parentFields =
-            if isStatic || decompiledClass.TypeInfo.Base = null then []
-            else getDefaultFieldValuesOf false decompiledClass.TypeInfo.Base.Type.AssemblyQualifiedName
+            if withParent
+            then
+                if isStatic || decompiledClass.TypeInfo.Base = null then [] else getDefaultFieldValuesOf false withParent decompiledClass.TypeInfo.Base.Type.AssemblyQualifiedName
+            else []
```

Comment:
if not withParent || isStatic || decompiledClass.TypeInfo.Base = null then []
    else getDefaultFieldValuesOf false withParent decompiledClass.TypeInfo.Base.Type.AssemblyQualifiedName